### ISSUE TITLE
umount can't be done in same Terminal on MacOS

### ISSUE THREAD

**Issue by novski** [restic/restic#815]
## Output of `restic version`
restic 0.4.0
compiled with go1.7.5 on darwin/amd64
## Expected behavior
After a mount im not able to give any command to the same terminal window, until i escape with CTRL&C wich interrupts, and this is the output:   Interrupt received, cleaning up

## Actual behavior
I wold expect terminal to let me enter a umount. But i can't execute anything.
(please let me know about the correct umount command as it lacks in the manual...). 
If i do with: diskutil umount /path/to/bakup in a separate window of terminal, it prints this in the main window: 
unable to umount (maybe already umounted?): unmount /Users/USERNAME/mnt/restic: invalid argument.

## Steps to reproduce the behavior
mount with:
restic -r /Volumes/Backup/restic mount ~/mnt/restic
try to unmount in same terminal window...

**Comment by fd0** [comment]
Good point, this is indeed missing in the manual. When you start restic with the `umount` command, you should be able to access the data stored in the repo at the given mountpoint location (`~/mnt/restic` in your case), e.g. by using the finder or a second terminal. While restic is running it servers the content there.

When you press ^C, restic umounts properly and then terminates. If you use an external command for umount, restic also detects that, tries to umount (which fails) and then terminates.

If you want to use the same shell to execute commands afterwards, you can press `^Z` to interrupt restic and then use the `bg` command to continue running restic in the background of the shell. Since this is standard shell behavior, we're not explaining this in the manual.

So, apart from a hint in the manual on how to properly umount I don't think there's anything we should do here.

What do you think?

**Comment by quantonganh** [comment]
> When you start restic with the umount command,

I think you meant to say `mount`.

> you should be able to access the data stored in the repo at the given mountpoint location (~/mnt/restic in your case), e.g. by using the finder or a second terminal.

OK.

> While restic is running it servers the content there.

Should/Can it returns the command prompt as normal instead of:

```
Now serving the repository at /tmp/lacie
Don't forget to umount after quitting!
```

and hang there?

**Comment by fd0** [comment]
Thanks for adding a comment. Yeah, we can build that. Since many people stumble upon this issue I think it worth implementing.

**Comment by wyattjoh** [comment]
Same tripped me up too.

It read to me as:

```sh
$ restic mount /mnt/restic
repository <...> opened successfully, password is correct
Now serving the repository at /mnt/restic
Don't forget to umount after quitting!

  signal interrupt received, cleaning up
$ umount /mnt/restic/
umount: /mnt/restic/: not mounted.
```

I assumed that I was supposed to umount it myself after I exit the command with CTRL-C.

Better I suppose to instead write something like:

```sh
$ restic mount /mnt/restic
repository <...> opened successfully, password is correct
Now serving the repository at /mnt/restic
Don't forget to umount or ctrl-c

  signal interrupt received
  cleaning up
  umount successful
$
```

**Comment by rawtaz** [comment]
Fixed in #2153, closing this issue.
